openbas-framework/src/main/java/io/openbas/utils/FilterUtilsJpa.java

```diff
@@ -111,6 +113,40 @@ private static <U> Predicate toPredicate(
     return operation.apply(paths, filter.getValues());
   }
 
+  /**
+   * Create a "in" specification for searches
+   *
+   * @param fieldName the JPA field on which the in rule is based
+   * @param inValues the values to include in the search for given field
+   * @param <T> the data type of the specification (usually a JPA entity)
+   * @return the built JPA Specification
+   */
+  public static <T> Specification<T> computeIn(
+      @Nullable final String fieldName, @Nullable final List<String> inValues) {
+    if (!hasText(fieldName) || CollectionUtils.isEmpty(inValues)) {
+      //noinspection unchecked
+      return (Specification<T>) EMPTY_SPECIFICATION;
+    }
+    return (root, query, cb) -> root.get(fieldName).in(inValues);
+  }
+
+  /**
+   * Create a "not in" specification for searches
+   *
+   * @param fieldName the JPA field on which the exclusion rule is based
+   * @param excludedValues the values to exclude from the search in given field
+   * @param <T> the data type of the specification (usually a JPA entity)
+   * @return the built JPA Specification
+   */
+  public static <T> Specification<T> computeNotIn(
```

Comment:
Can be modified to : Specification.not(computeIn(fieldName, excludedValues))

Reply:
Updated